lib/shelley/test/unit/Cardano/Wallet/Shelley/CompatibilitySpec.hs

```diff
@@ -309,6 +322,190 @@ spec = do
         forM_ matrix $ \(title, addr, predicate) ->
             it title $ inspectAddress addr `shouldSatisfy` predicate
 
+    describe "golden tests for script hashes for different versions" $ do
+        testScriptsAllLangs Cardano.SimpleScriptV1
+        testScriptsAllLangs Cardano.SimpleScriptV2
+        testScriptsTimelockLang
+
+    describe "golden tests for script preimages for different versions" $ do
+        testScriptPreimages Cardano.SimpleScriptV1
+        testScriptPreimages Cardano.SimpleScriptV2
+        testTimelockScriptImagesLang
+
+toKeyHash :: Text -> Script KeyHash
+toKeyHash txt = case fromBase16 (T.encodeUtf8 txt) of
+    Right bs -> case keyHashFromBytes bs of
+        Just kh -> RequireSignatureOf kh
+        Nothing -> error "Hash key not valid"
+    Left _ -> error "Hash key not valid"
+
+toPaymentHash :: Text -> Cardano.SimpleScript lang
+toPaymentHash txt =
+    case Cardano.deserialiseFromRawBytesHex (Cardano.AsHash Cardano.AsPaymentKey) (T.encodeUtf8 txt) of
+        Just payKeyHash -> Cardano.RequireSignature payKeyHash
+        Nothing -> error "Hash key not valid"
+
+checkScriptHashes
+    :: String
+    -> Script KeyHash
+    -> Cardano.Script lang
+    -> SpecWith ()
+checkScriptHashes title adrestiaScript nodeScript = it title $
+    unScriptHash (toScriptHash adrestiaScript) `shouldBe`
+    Cardano.serialiseToRawBytes (Cardano.hashScript nodeScript)
+
+checkScriptPreimage
+    :: Cardano.SerialiseAsCBOR (Cardano.Script lang)
+    => String
+    -> Script KeyHash
+    -> Cardano.Script lang
+    -> SpecWith ()
+checkScriptPreimage title adrestiaScript nodeScript = it title $
+    (serializeScript adrestiaScript) `shouldBe`
+    BS.append "\00" (Cardano.serialiseToCBOR nodeScript)
+
+scriptMatrix
+    :: Cardano.SimpleScriptVersion lang
+    -> [(String, Script KeyHash, Cardano.Script lang)]
+scriptMatrix version =
+    [ ( show version <> " RequireSignatureOf"
```

Comment:
We can probably drop the `show version` bit now because the `describe` already has it.